File: src/SangYeop/15649.py
n,m = map(int,input().split())

arr = []

def BT():
  if len(arr)==m:
    print(' '.join(map(str,arr)))
    return
  for i in range(1,n+1):
    if i not in arr:
      arr.append(i)
      BT()
      arr.pop()

BT()

File: src/SangYeop/15649.js
// const input = require('fs').readFileSync('/dev/stdin').toString().split('\n').slice(0,-1);
const input = require('fs').readFileSync('example.txt').toString().trim().split('\n');
const NM = input[0].split(" ");
const [n, m] = NM.map(el => Number(el));

const sequence = [...Array(m)].fill(0);
const visited = [...Array(n+1)].fill(false);
let result = ''

function dfs(k) {
  if (k === m) {
    const arr = [];
    for (let i = 0; i < m; i++){
      arr.push(sequence[i]);
    }
    return result += `${arr.join(' ')}\n`;
  }

  for (let i = 1; i <= n; i++){
    if (!visited[i]) {
      sequence[k] = i;
      visited[i] = true;
      dfs(k + 1);
      visited[i] = false;
    }
  }
}

dfs(0)
console.log(result)